### ISSUE TITLE
Add Edit Profile picture functionality in the Administrator Controls [Blocked on mocks]

### ISSUE THREAD

**Issue by FareesHussain** [oppia/oppia-android#3132]
**Is your feature request related to a problem? Please describe.**
Add an **Edit Profile Picture** functionality in Administrator Controls which already includes the **rename** and **reset pin** functionality. So that the user (Admin) can change the profile picture for other users (non-admin) instead of needing to first log in the user and then update it via the profile screen from the drawer menu.

**Comment by rt4914** [comment]
@mschanteltc 
What are you thoughts about this?

**Comment by mschanteltc** [comment]
I don't believe we should give Administrator's the ability to change a user's profile picture. Those are usually personalized and any inappropriate uploads should be discussed outside of the app. The administrator has the ability to delete the profile if they think it is necessary. I don't believe there are any other applications that allow Administrator-like users to change other profile pictures through the settings.

**Comment by FareesHussain** [comment]
@mschanteltc I just added this as a UX suggestion while testing. Shall I close this?

**Comment by BenHenning** [comment]
I think @mschanteltc's conclusion makes sense. @FareesHussain for context, what led you to thinking this feature should exist? Just because it's one piece of functionality that admins can't do?

**Comment by FareesHussain** [comment]
I was thinking of a use case where the admin wants to change all users profile pictures and as the admin can rename or change the passwords for other users, I thought It would be better to include profile picture too.

**Comment by mschanteltc** [comment]
Now that you mention it, renaming profiles from the settings is not common either, though we do not have that as an option in the [User Profile](https://xd.adobe.com/view/3dca36c2-5115-419c-b25e-0f10526b077c-6899/screen/7fb5929e-4652-4412-bf8c-64d94f4159f9/) page. We can move name editing to the User Profile page to make it consistent. 

The Administrator should still be able to change PINs though, mostly because it is a privacy/security reasons rather than user profile personalization. WDYT?

**Comment by FareesHussain** [comment]
> Now that you mention it, renaming profiles from the settings is not common either, though we do not have that as an option in the [User Profile](https://xd.adobe.com/view/3dca36c2-5115-419c-b25e-0f10526b077c-6899/screen/7fb5929e-4652-4412-bf8c-64d94f4159f9/) page. We can move name editing to the User Profile page to make it consistent.
> 
> The Administrator should still be able to change PINs though, mostly because it is a privacy/security reasons rather than user profile personalization. WDYT?

Sounds good to me.

**Comment by rt4914** [comment]
@mschanteltc 
These are some of the concerns/suggestions that came from team meeting:
1. We should shift the rename as well as reset the pin to child-profile accessible from the navigation drawer. 
2. The reset pin should not be a security concern as the child has logged in via pin-verification screen which already proves that the profile is correct.
3. Concern with above solution: If we allow rename in child-profile and there is a case where a teacher and students are using this app then the student might change the name / profile picture to something which is not appropriate and therefore only admin profiles should be allowed to change picture/name/pin.

WDYT?

**Comment by mschanteltc** [comment]
> @mschanteltc
> These are some of the concerns/suggestions that came from team meeting:
> 
> 1. We should shift the rename as well as reset the pin to child-profile accessible from the navigation drawer.
> 2. The reset pin should not be a security concern as the child has logged in via pin-verification screen which already proves that the profile is correct.
> 3. Concern with above solution: If we allow rename in child-profile and there is a case where a teacher and students are using this app then the student might change the name / profile picture to something which is not appropriate and therefore only admin profiles should be allowed to change picture/name/pin.
> 
> WDYT?

Wouldn't it be a security concern if the user initially does not have a PIN set and a PIN is created through their profile? We would have no way of ensuring the user who set the PIN is that user. If we only let users who already have a PIN to change their PIN (meaning users who do not have a PIN can only be made through the Admin settings), would we also give them an option to remove a PIN?

Maybe instead of allowing all users rename/profile change abilities, we only allow it for users who have "Download access" available to them. The drawback is that this feature is not implied with the name "Download Access" and users who do not have access to this are not allowed to change their name or profile picture. WDYT?

**Comment by rt4914** [comment]
> > @mschanteltc
> > These are some of the concerns/suggestions that came from team meeting:
> > 
> > 1. We should shift the rename as well as reset the pin to child-profile accessible from the navigation drawer.
> > 2. The reset pin should not be a security concern as the child has logged in via pin-verification screen which already proves that the profile is correct.
> > 3. Concern with above solution: If we allow rename in child-profile and there is a case where a teacher and students are using this app then the student might change the name / profile picture to something which is not appropriate and therefore only admin profiles should be allowed to change picture/name/pin.
> > 
> > WDYT?
> 
> Wouldn't it be a security concern if the user initially does not have a PIN set and a PIN is created through their profile? We would have no way of ensuring the user who set the PIN is that user. If we only let users who already have a PIN to change their PIN (meaning users who do not have a PIN can only be made through the Admin settings), would we also give them an option to remove a PIN?
> 
This makes sense. We can implement it like this: (a.) if child profile does not have a pin, in [ProfileProgress page](https://xd.adobe.com/view/3dca36c2-5115-419c-b25e-0f10526b077c-6899/screen/7fb5929e-4652-4412-bf8c-64d94f4159f9/) we do not show `reset pin` option. (b.) if child profile does have a pin in ProfileProgress page we do show them `reset pin` option. Does this resonate with what you are mentioning?

> Maybe instead of allowing all users rename/profile change abilities, we only allow it for users who have "Download access" available to them. The drawback is that this feature is not implied with the name "Download Access" and users who do not have access to this are not allowed to change their name or profile picture. WDYT?

Yes it does not make direct sense because "Download Access" text but basically the general idea is that if the child profile has `download access` it means that the admin has given them extra privileges and therefore they can edit profile information. But how can we easily communicate this to the user?

@mschanteltc

**Comment by mschanteltc** [comment]
> > > @mschanteltc
> > > These are some of the concerns/suggestions that came from team meeting:
> > > 
> > > 1. We should shift the rename as well as reset the pin to child-profile accessible from the navigation drawer.
> > > 2. The reset pin should not be a security concern as the child has logged in via pin-verification screen which already proves that the profile is correct.
> > > 3. Concern with above solution: If we allow rename in child-profile and there is a case where a teacher and students are using this app then the student might change the name / profile picture to something which is not appropriate and therefore only admin profiles should be allowed to change picture/name/pin.
> > > 
> > > WDYT?
> > 
> > 
> > Wouldn't it be a security concern if the user initially does not have a PIN set and a PIN is created through their profile? We would have no way of ensuring the user who set the PIN is that user. If we only let users who already have a PIN to change their PIN (meaning users who do not have a PIN can only be made through the Admin settings), would we also give them an option to remove a PIN?
> 
> This makes sense. We can implement it like this: (a.) if child profile does not have a pin, in [ProfileProgress page](https://xd.adobe.com/view/3dca36c2-5115-419c-b25e-0f10526b077c-6899/screen/7fb5929e-4652-4412-bf8c-64d94f4159f9/) we do not show `reset pin` option. (b.) if child profile does have a pin in ProfileProgress page we do show them `reset pin` option. Does this resonate with what you are mentioning?
> 
Yes that is correct. 
> > Maybe instead of allowing all users rename/profile change abilities, we only allow it for users who have "Download access" available to them. The drawback is that this feature is not implied with the name "Download Access" and users who do not have access to this are not allowed to change their name or profile picture. WDYT?
> 
> Yes it does not make direct sense because "Download Access" text but basically the general idea is that if the child profile has `download access` it means that the admin has given them extra privileges and therefore they can edit profile information. But how can we easily communicate this to the user?
> 
> @mschanteltc

I'm thinking we can change the toggle to "Kids Profile" to connote limited access and features for the user. The description can be changed to "Disable download access and profile customization (Changing PIN, profile picture, and name)." WDYT?

**Comment by rt4914** [comment]
@mschanteltc Makes sense. Thanks. Once mocks are ready please let me know. Although this is not on high priority right now.
